### ISSUE TITLE
SSE Response.Flush

### ISSUE THREAD

**Issue by Nness** [ServiceStack/Issues#379]
I'm just curious on SSE's implementation when using Response.Flush. I understand that ServiceStack currently is .Net 4.0 based. Are there any way I can use Response.FlushAsync inside SSE feature?

As our SSE server getting more and more user, we have performance hit on CPU. One of assumption is when connection closed by client, server side flush cause exception to popup, that take time and also blocks CPU. Thus async might be able to help here.

**Comment by mythz** [comment]
ServiceStack is a .NET 4.0 assembly which doesn't have access to Write/Flush Async methods but I've made the Write and Flush overridable with the new `ServerEventsFeature.WriteEvent` which you can override to use async equivalents in a .NET 4.5 project with something like:

```csharp
Plugins.Add(new ServerEventsFeature
{
    WriteEvent = (res, frame) =>
    {
        var aspRes = (HttpResponseBase)res.OriginalResponse;
        var bytes = frame.ToUtf8Bytes();
        aspRes.OutputStream.WriteAsync(bytes, 0, bytes.Length)
            .Then(_ => aspRes.OutputStream.FlushAsync());
    }
});
```

This change is available from v4.0.51 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet). Please let me know if this helps at all and the number of connections where it starts to be an issue?

In future, please reserve this issues list for just reporting reproducible bugs/issues, you can discuss other technical support questions in the Customer Forums: https://forums.servicestack.net

**Comment by Nness** [comment]
Sure. thx.
